src/code/z_actor.c

```diff
@@ -457,439 +1231,3485 @@ f32 Actor_XZDistanceToPoint(Actor* actor, Vec3f* point) {
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    POLY_OPA_DISP = Gfx_CallSetupDL(POLY_OPA_DISP, 0x2C);
+    gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 0, 0, 0, arg2);
+
+    sp48.x = arg0->x;
+    sp48.y = arg0->y + 1.0f;
+    sp48.z = arg0->z;
+
+    sp54 = BgCheck_EntityRaycastFloor2(globalCtx, &globalCtx->colCtx, &sp44, &sp48);
+    if (sp44 != NULL) {
+        func_800C0094(sp44, arg0->x, sp54, arg0->z, &sp58);
+        Matrix_SetCurrentState(&sp58);
+    } else {
+        Matrix_InsertTranslation(arg0->x, arg0->y, arg0->z, MTXMODE_NEW);
+    }
+    Matrix_Scale(arg1->x, 1.0f, arg1->z, MTXMODE_APPLY);
+
+    gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPDisplayList(POLY_OPA_DISP++, D_04076BC0);
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
+}
+
+void func_800BC770(GameState* gameState, s16 arg1, s16 arg2) {
+    s16 idx;
+    GlobalContext* globalCtx = (GlobalContext*)gameState;
+
+    idx = Quake_Add(&globalCtx->mainCamera, 3);
+    Quake_SetSpeed(idx, 20000);
+    Quake_SetQuakeValues(idx, arg1, 0, 0, 0);
+    Quake_SetCountdown(idx, arg2);
+}
+
+void func_800BC7D8(GameState* gameState, s16 arg1, s16 arg2, s16 speed) {
+    s16 idx;
+    GlobalContext* globalCtx = (GlobalContext*)gameState;
+
+    idx = Quake_Add(&globalCtx->mainCamera, 3);
+    Quake_SetSpeed(idx, speed);
+    Quake_SetQuakeValues(idx, arg1, 0, 0, 0);
+    Quake_SetCountdown(idx, arg2);
+}
+
+void func_800BC848(Actor* actor, GameState* gameState, s16 arg2, s16 arg3) {
+    if (arg2 >= 5) {
+        func_8013ECE0(actor->xyzDistToPlayerSq, 0xFF, 0x14, 0x96);
+    } else {
+        func_8013ECE0(actor->xyzDistToPlayerSq, 0xB4, 0x14, 0x64);
+    }
+    func_800BC770(gameState, arg2, arg3);
+}
+
+extern Gfx D_05000140[];
+extern Gfx D_05000230[];
+extern Gfx D_06000530[];
+extern Gfx D_06000400[];
+
+typedef struct {
+    /* 0x00 */ f32 unk_00;
+    /* 0x04 */ f32 unk_04;
+    /* 0x08 */ f32 unk_08;
+    /* 0x0C */ f32 unk_0C;
+    /* 0x10 */ f32 unk_10;
+    /* 0x14 */ Gfx* unk_14;
+    /* 0x18 */ Gfx* unk_18;
+} struct_801AEDD4; // size = 0x1C
+
+struct_801AEDD4 D_801AEDD4[] = {
+    { 0.540000021458f, 6000.0f, 5000.0, 1.0f, 0.0f, D_05000230, D_05000140 },
+    { 0.643999993801f, 12000.0f, 8000.0f, 1.0f, 0.0f, D_06000530, D_06000400 },
+    { 0.6400000453f, 8500.0f, 8000.0f, 1.75f, 0.1f, D_05000230, D_05000140 },
+};
+
+void Actor_DrawDoorLock(GlobalContext* globalCtx, s32 frame, s32 type) {
+    s32 pad[2];
+    MtxF spA8;
+    s32 i;
+    f32 sin;
+    f32 cos;
+    struct_801AEDD4* entry = &D_801AEDD4[type];
+    f32 phi_f20;
+    f32 phi_f2;
+
+    phi_f20 = entry->unk_10;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    Matrix_InsertTranslation(0.0f, entry->unk_08, 500.0f, MTXMODE_APPLY);
+    Matrix_CopyCurrentState(&spA8);
+
+    sin = __sinf(entry->unk_00 - phi_f20) * -(10 - frame) * 0.1f * entry->unk_04;
+    cos = __cosf(entry->unk_00 - phi_f20) * (10 - frame) * 0.1f * entry->unk_04;
```

Comment:
What do you think about using temps called `sin` and `cos`. I've certainly used temps named this before, but is it too close to some intrinsic function? Or am I overthinking?

Reply:
We have used those names for other stuff in other places, so I don't think it is a big deal.
I would like to revisit this function and improve its documentation a bit tho